src/main/java/org/kuali/kfs/sys/context/BatchStepExecutor.java

```diff
@@ -0,0 +1,243 @@
+ * 
+ * BatchStepExecutor executes a Step in its own Thread and writes either a .success or .error file after execution.
+ * This class notifies the ContainerStepListener when a Step has started and when it has completed.
+ * <p>
+ * BatchStepExecutor adds a ConsoleAppender to its Logger if one hasn't been configured.
+ */
+public class BatchStepExecutor implements Runnable {
+
+    private static final Logger LOG = LogManager.getLogger();
+
+    private final ParameterService parameterService;
+    private final DateTimeService dateTimeService;
+    private final BatchContainerDirectory batchContainerDirectory;
+    private final BatchStepFileDescriptor batchStepFile;
+    private final Step step;
+    private final int stepIndex;
+
+    private String logFileName;
+
+    private final List<ContainerStepListener> containerStepListeners;
+
+    /**
+     * @param parameterService        the ParameterService used by Job
+     * @param dateTimeService         the DateTimeService used by Job
+     * @param batchContainerDirectory the batch container directory
+     * @param batchStepFile           the descriptor containing information about the step to execute
+     * @param step                    the Step to execute
+     * @param stepIndex               the index of the step in the job
+     */
+    public BatchStepExecutor(
+            final ParameterService parameterService, final DateTimeService dateTimeService,
+            final BatchContainerDirectory batchContainerDirectory, final BatchStepFileDescriptor batchStepFile, final Step step,
+            final int stepIndex) {
+        this.parameterService = parameterService;
+        this.dateTimeService = dateTimeService;
+
+        this.batchContainerDirectory = batchContainerDirectory;
+        this.batchStepFile = batchStepFile;
+        this.step = step;
+        this.stepIndex = stepIndex;
+
+        containerStepListeners = new ArrayList<>();
+
+        LOG.info("Initialized thread executor for {}", batchStepFile);
+    }
+
+    /**
+     * Execute the Step via Job.runStep(). Setup NDC logging so the Step has its own log file. Remove the NDC logging
+     * once the step is finished executing. Notify the ContainerStepListeners when the step starts and finishes.
+     */
+    @Override
+    public void run() {
+        final Date stepRunDate = dateTimeService.getCurrentDate();
+        batchStepFile.setStartedDate(stepRunDate);
+        batchStepFile.setStepIndex(stepIndex);
+
+        try (CloseableThreadContext.Instance ignored = setupNDCLogging()) {
+            runWithLogging(stepRunDate);
+            /*
+             * CU Customization: Copied the contents of the "catch" block below this line into the runWithLogging()
+             * method instead, and replaced the original contents below with warning-only error handling. 
+             */
+        } catch (final Exception throwable) {
+            LOG.warn(
```

Comment:
should these be error log messages rather than warnings? Also, should the full error be logged, rather than just the message?

Reply:
I'm not sure why, but KualiCo had set this part up so that the error would be logged as a warning and would only show partial details here. I was trying to be consistent with how the `BatchStepExecutor` was currently logging things.

Also, the downstream code in `Job.runStep()` should already be logging the exception's full stack trace at error level.